src/linker/pipeline_schema.py

```diff
@@ -18,26 +18,28 @@ def get_schemas(cls) -> List["PipelineSchema"]:
         """Creates the allowable schema for the pipeline."""
         schemas = []
         pvs_like_case_study = cls("pvs_like_case_study")
-        pvs_like_case_study.add_step(Step("pvs_like_case_study"))
+        # Add the steps in the order they should be run
+        pvs_like_case_study._add_step(Step("pvs_like_case_study"))
         schemas.append(pvs_like_case_study)
         return schemas
 
-    def add_step(self, step: Step):
+    def _add_step(self, step: Step):
         self.steps.append(step)
 
     def validate_pipeline(self, pipeline: Pipeline):
-        # TODO:
-        ## ensure implementation is assigned to the correct step
-        ## ensure that the implementations are in the correct step order
+        # Loop through the pipeline implementations and check if their order
+        ## and corresponding steps match the schema
+        for idx, implementation in enumerate(pipeline.implementations):
+            if implementation.step != self.steps[idx]:
+                return False
         return True
 
 
-_PIPELINE_SCHEMAS = PipelineSchema.get_schemas()
-
-
 def validate_pipeline(pipeline: Pipeline):
     """Validates the pipeline against supported schemas."""
-    for schema in _PIPELINE_SCHEMAS:
+    for schema in PipelineSchema.get_schemas():
         if schema.validate_pipeline(pipeline):
```

Comment:
Rajan: switch order to be `pipeline.validate(schema)`. By reversing this order, it should be simpler to validate b/c we know that a pipeline is a list (as opposed to schemas being potentially more complicated)